frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -645,60 +645,87 @@ export function ContainerInfo(props: ContainerInfoProps) {
+      let startedDate = state?.running?.startedAt || state?.terminated?.startedAt || '';
+      if (!!startedDate) {
+        startedDate = localeDate(startedDate);
+      }
+      return startedDate;
     }
 
-    let finishDate = status?.state?.terminated?.finishedAt ?? '';
-    if (!!finishDate) {
-      finishDate = localeDate(finishDate);
+    function getFinishDate(state?: KubeContainerStatus['state']) {
+      let finishDate = state?.terminated?.finishedAt || '';
+      if (!!finishDate) {
+        finishDate = localeDate(finishDate);
+      }
+      return finishDate;
     }
 
-    return [startedDate, finishDate];
+    return [
+      getStartedDate(status?.state),
+      getFinishDate(status?.state),
+      getStartedDate(status?.lastState),
+      getFinishDate(status?.lastState),
+    ];
   }, [status]);
 
-  function getContainerStatusLabel() {
-    if (!status || !container) {
-      return undefined;
-    }
+  function ContainerStatusLabel(props: {
+    state?: KubeContainerStatus['state'] | null;
+    container?: KubeContainer;
+  }) {
+    const { state, container } = props;
+
+    const [stateDetails, label, statusType] = React.useMemo(() => {
+      let stateDetails: KubeContainerStatus['state']['waiting' | 'terminated'] | null = null;
+      let label = t('translation|Ready');
+      let statusType: StatusLabelProps['status'] = '';
 
-    let state: KubeContainerStatus['state']['waiting' | 'terminated'] | null = null;
-    let label = t('translation|Ready');
-    let statusType: StatusLabelProps['status'] = '';
-
-    if (!!status.state.waiting) {
-      state = status.state.waiting;
-      statusType = 'warning';
-      label = t('translation|Waiting');
-    } else if (!!status.state.running) {
-      statusType = 'success';
-      label = t('translation|Running');
-    } else if (!!status.state.terminated) {
-      if (status.state.terminated.exitCode === 0) {
-        statusType = '';
-        label = status.state.terminated.reason;
-      } else {
-        statusType = 'error';
-        label = t('translation|Error');
+      if (!state) {
+        return [stateDetails, label, statusType];
       }
-    }
 
-    const tooltipID = 'container-state-message-' + container.name;
+      if (!!state.waiting) {
+        stateDetails = state.waiting;
+        statusType = 'warning';
+        label = t('translation|Waiting');
+      } else if (!!state.running) {
+        statusType = 'success';
+        label = t('translation|Running');
+      } else if (!!state.terminated) {
+        stateDetails = state.terminated;
+        if (state.terminated.exitCode === 0) {
+          statusType = '';
+          label = state.terminated.reason;
+        } else {
+          statusType = 'error';
+          label = t('translation|Error');
+        }
+      }
+
+      return [stateDetails, label, statusType];
+    }, [state]);
+
+    const tooltipID = React.useMemo(
+      () => 'container-state-message-' + (container?.name ?? ''),
```

Comment:
useMemo is a bit overkill for this, it's not that expensive

Reply:
You are right. I got excited.